.github/actions/override-workflow-variables/action.yml

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Override Workflow Variables'
+description: |-
+  Use this action to load/override variables into GITHUB_ENV.
+
+inputs:
+  working_directory:
+    description: 'The working directory to look for the env file.'
+    required: false
+    default: '.'
+  env_file:
+    description: 'The env file containing variables to override with.'
+    required: false
+    default: '.env'
+  require_env:
```

Comment:
Can you help me understand what this is for / why it's needed? I understand the point of loading an .env file, but I don't understand the reason for the the other stuff to enforce the presence of files that are never loaded.

Reply:
I believe the idea is that you may want to always load an env file for a specific workflow and this would prevent misconfiguration. The specific use case was guardian, if we have environment folders, we want to make sure each environment has its own file.

Reply:
This is useful for situations when workflow variables with the same name are reused across different folders. If two folders within a repository belong to two separate product teams, we may need to enforce `require_env=true` on their parent folders to ensure they are using separate service accounts to perform actions for their product only. 

in short, we can share the same workflow but keep separation between product folders/teams.